contracts/infinite.cairo

```diff
@@ -0,0 +1,338 @@
+%lang starknet
+%builtins pedersen range_check bitwise
+
+from starkware.cairo.common.cairo_builtins import (HashBuiltin,
+    BitwiseBuiltin)
+from starkware.cairo.common.math import (assert_not_zero, 
+    assert_le_felt, assert_not_equal)
+from starkware.starknet.common.syscalls import get_caller_address
+from starkware.cairo.common.memcpy import memcpy
+from starkware.cairo.common.alloc import alloc
+
+from contracts.utils.packing import pack_game, unpack_game
+from contracts.utils.life_rules import evaluate_rounds
+from contracts.utils.constants import DIM
+
+
+##### Storage #####
+
+# Returns the generation of the current alive generation.
+@storage_var
+func current_generation() -> (generation : felt):
+end
+
+# Returns the user_id for a given generation.
+@storage_var
+func owner_of_generation(generation : felt) -> (user_id : felt):
+end
+
+# Returns the total number of credits owned by a user.
+@storage_var
+func credits(user_id : felt) -> (credits : felt):
+end
+
+# Stores cells revival history for user in (cell_index, generation) tuple
+@storage_var
+func revival_history(
```

Comment:
1. This stores only one revival, so it is not a history.
2. Why do we need this on chain?